init.d/02.setup-resources/manifests/frontend/game.js

```diff
@@ -231,16 +262,63 @@ function render() {
 function startGame() {
     hideEndscreen();
     hideStartscreen();
+    gameState.name = document.getElementById("nameInput").value;
     gameState.running = true;
     gameState.timer.start();
 }
 
+function checkAlreadyPlaying() {
+    const gameStateString = localStorage.getItem("activeGame");
+    if (gameStateString) {
+        const oldGameState = JSON.parse(gameStateString);
+        Object.assign(gameState, oldGameState);
+        Object.setPrototypeOf(oldGameState.timer, Timer.prototype);
+        document.getElementById("nameInput").value = gameState.name;
+        startGame();
+    }
+}
+
 function stopGame() {
     gameState.running = false;
     gameState.timer.stop();
     showEndscreen();
 }
 
+function getHighscores() {
+    const highscores = localStorage.getItem("highscores");
+    if (!highscores) {
+        return [];
+    }
+    return JSON.parse(highscores);
+}
+
+function saveHighscores(highscores) {
+    localStorage.setItem("highscores", JSON.stringify(highscores));
+}
+
+function addHighscore() {
+    const entry = [gameState.name, gameState.timer.getTime()];
+    const highscores = getHighscores();
+    highscores.push(entry);
+    saveHighscores(highscores);
+}
+
+function showHighscores() {
+    let highscores = getHighscores();
+    highscores.sort((a, b) => a[1].localeCompare(b[1]));
```

Comment:
It's functionally the same, but I'd suggest a list of objects to make it easier to understand

```suggestion
    const entry = { name: gameState.name, score: gameState.timer.getTime() };
    const highscores = getHighscores();
    highscores.push(entry);
    saveHighscores(highscores);
}

function showHighscores() {
    const highscores = getHighscores();
    highscores.sort((a, b) => a.score - b.score);
```

If substraction on strings sounds too dangerous to you (JavaScript will convert strings to numbers in substraction, but not in addition...), we could also update the timer to have `getTime () => number` and `toString () => string`.

Reply:
Changed to now to compare seconds.